redisinsight/ui/src/slices/rdi/instances.ts

```diff
@@ -0,0 +1,297 @@
+    defaultInstanceChangingFailure: (state, { payload = '' }) => {
+      state.loadingChanging = false
+      state.changedSuccessfully = false
+      state.errorChanging = payload.toString()
+    },
+
+    // test instance connection
+    testConnection: (state) => {
+      state.loadingChanging = true
+      state.errorChanging = ''
+    },
+    testConnectionSuccess: (state) => {
+      state.loadingChanging = false
+    },
+    testConnectionFailure: (state, { payload = '' }) => {
+      state.loadingChanging = false
+      state.errorChanging = payload.toString()
+    },
+
+    // delete instances
+    setDefaultInstance: (state) => {
+      state.loading = true
+      state.error = ''
+    },
+    setDefaultInstanceSuccess: (state) => {
+      state.loading = false
+    },
+    setDefaultInstanceFailure: (state, { payload }) => {
+      state.loading = false
+      state.error = payload
+    },
+
+    // set connected instance id
+    setConnectedInstanceId: (state, { payload }: { payload: string }) => {
+      state.connectedInstance = {
+        ...state.connectedInstance,
+        id: payload
+      }
+    },
+
+    // set edited instance
+    setEditedInstance: (state, { payload }: { payload: Nullable<RdiInstance> }) => {
+      state.editedInstance.data = payload
+    },
+
+    // reset connected instance
+    resetConnectedInstance: (state) => {
+      state.connectedInstance = initialState.connectedInstance
+    }
+  }
+})
+
+// Actions generated from the slice
+export const {
+  loadInstances,
+  loadInstancesSuccess,
+  loadInstancesFailure,
+  defaultInstanceChanging,
+  defaultInstanceChangingSuccess,
+  defaultInstanceChangingFailure,
+  testConnection,
+  testConnectionSuccess,
+  testConnectionFailure,
+  setDefaultInstance,
+  setDefaultInstanceSuccess,
+  setDefaultInstanceFailure,
+  setConnectedInstanceId,
+  setEditedInstance,
+  resetConnectedInstance
+} = instancesSlice.actions
+
+// selectors
+export const instancesSelector = (state: RootState) => state.rdi.instances
+
+// The reducer
+export default instancesSlice.reducer
+
+// Asynchronous thunk action
+export function fetchInstancesAction(onSuccess?: (data?: RdiInstanceResponse[]) => void) {
+  return async (dispatch: AppDispatch) => {
+    dispatch(loadInstances())
+
+    try {
+      const data: RdiInstanceResponse[] = []
+
+      const status = 200
+
+      if (isStatusSuccessful(status)) {
+        onSuccess?.(data)
+        dispatch(loadInstancesSuccess(data))
+      }
+    } catch (_err) {
+      const error = _err as AxiosError
+      const errorMessage = getApiErrorMessage(error)
+      dispatch(loadInstancesFailure(errorMessage))
+      dispatch(addErrorNotification(error))
+    }
+  }
+}
+
+// Asynchronous thunk action
+export function createInstancesAction(onSuccess?: (data?: RdiInstanceResponse[]) => void) {
+  return async (dispatch: AppDispatch) => {
+    dispatch(loadInstances())
+
+    try {
+      const data: RdiInstanceResponse[] = [
+        {
```

Comment:
Please use comments "// mock response" or "// TODO remove it after be implementation
Or you can merge backend part and implement new endpoint